### PR TITLE
Adding destructors for leak safety

### PR DESCRIPTION
**Pull request by fherzog2**

### CODE REVIEW

File: image/image.h [hunk 1]
```diff
@@ -55,7 +60,63 @@ class Image {
...
+      other.depth = 0;
+      
+      palette = other.palette;
+      col_begin = std::move(other.col_begin);
+      col_end = std::move(other.col_end);
+      seen_before = other.seen_before;
+      
+      other.palette = false;
+      other.seen_before = 0;
+	}
```
Comment (hrj) [inline image/image.h]: The above change gives me this warning:

```
In file included from image/image.cpp:4:
image/image.h:118:2: warning: control reaches end of non-void function [-Wreturn-type]
        }
        ^
```

Not sufficiently familiar with C++ to fix it myself.

**Comment by hrj** [comment]
Alert: This PR has introduced a seg-fault when decoding truncated files.

**Comment by jonsneyers** [comment]
Oops. We need more test cases I guess. Will be afk for a while.

**Comment by hrj** [comment]
Yeah, I am adding a truncated-file test now.

**Comment by hrj** [comment]
Whoa, look at the [logs in the travis build](https://travis-ci.org/jonsneyers/FLIF/builds/83794420)

It has a `Segmentation fault (core dumped)`. So even normal files are core-dumping. Working on improving the tests!

**Comment by fherzog2** [comment]
That went well... I didn't get any errors during my tests. Maybe mingw is more forgiving.

**Pull request merged**